pkg/controller/validation/account_validation_controller.go

```diff
@@ -150,6 +166,35 @@ func MoveAccount(awsAccountId string, client awsclient.Client, targetOU string,
 	return nil
 }
 
+// ValidateAccountTags avulaj: accountTagEnabled can be used in the future if we decide we want to fix this issue as we come across it during validation
+func ValidateAccountTags(client awsclient.Client, accountId *string, shardName string, accountTagEnabled bool) error {
+	listTagsForResourceInput := &organizations.ListTagsForResourceInput{
+		ResourceId: accountId,
+	}
+
+	resp, err := client.ListTagsForResource(listTagsForResourceInput)
+	if err != nil {
+		return err
+	}
+
+	for _, tag := range resp.Tags {
+		if ownerKey == *tag.Key {
+			if shardName != *tag.Value {
```

Comment:
seems possible for shardName to be unset based on the logic above which gets it from the AAO config map:
	hiveName, ok := configMap.Data["shard-name"]
	if !ok {
		log.Error(err, "shard-name key not available in configmap")
	}
	reconciler.shardName = hiveName

The most likely reason for this would be misconfiguration, however, that misconfiguration would also cause a bunch of noise here. That could be considered a feature, but im wondering if we want to skip tag validation based on shard name if its nil/empty. It doesnt look like we are attempting to update the tag if its different (which would result in information loss if things are misconfigured) so this is more of a what do you think comment than a request.